colossalai/shardformer/shard/shard_config.py

```diff
@@ -46,6 +44,10 @@ def tensor_parallel_size(self):
         return self._tensor_parallel_size
 
     def __post_init__(self):
+        if self.extra_kwargs:
+            for key, value in self.extra_kwargs.items():
+                setattr(self, key, value)
```

Comment:
Is it essential to copy the attr back? If so you have to prevent conflicts here

Reply:
Removed